### PR TITLE
[csrng] Define, fix, and verify `cs_aes_halt`

### PR DESCRIPTION
**Pull request by andreaskurth**
This PR:
- Adds a minimal specification of CSRNG's behavior on the `cs_aes_halt` interface.
- Fixes the acknowledge output of `cs_aes_halt` driven by CSRNG to be level instead of pulse.
- Configures CSRNG's DV so that it does not deassert the request of `cs_aes_halt` after a single acknowledge cycle.
- Adds assertions to CSRNG RTL to ensure its AES is only active while `cs_aes_halt` is not.

Tested locally with a full regression run of CSRNG. The added assertion never failed and all test results are in line with current nightly results.

### CODE REVIEW

**Review by cindychip — APPROVED** [review]
LGTM, do we need to add any chip level test to cover this, or the block level assertion checks are enough? (Maybe it is already covered in some AES enable/disable tests?)

**Comment by andreaskurth** [comment]
Thanks for your feedback, @cindychip.

> do we need to add any chip level test to cover this, or the block level assertion checks are enough?

Both CSRNG and Entropy Source block-level DV have agents to randomly requests/acknowledge the `cs_aes_halt` interface. With this PR (and #17943 for Entropy Source), both blocks have RTL assertions that ensure they correctly implement their side of the interface. 

Top-level tests implicitly test this feature when they use the entropy complex; in particular the max power test should use both blocks intensively and concurrently. There currently is a top-level testpoint (`chip_sw_entropy_src_cs_aes_halt`) that would specifically test this feature, but Entropy Source has some limitations (see #17943). Those limitations will be covered by block-level assertions, but we think replicating these checks at the top level doesn't add value. So we plan to remove that TLT in #17951.

TL;DR: We're verifying this feature in both blocks with this PR and #17943 and remove the top-level testpoint in #17951.

**Review by vogelpi — APPROVED** [review]
Just two nits, but LGTM, thanks @andreaskurth !

Also the AES activity detection looks good.

File: hw/ip/csrng/rtl/csrng_core.sv [hunk 1]
```diff
@@ -1630,7 +1630,7 @@ module csrng_core import csrng_pkg::*; #(
   // es to cs halt request to reduce power spikes
   assign cs_aes_halt_d =
          (ctr_drbg_upd_es_ack && ctr_drbg_gen_es_ack && block_encrypt_quiet &&
-          cs_aes_halt_i.cs_aes_halt_req && !cs_aes_halt_q);
+          cs_aes_halt_i.cs_aes_halt_req);
```
[the remark this answers is not in the document]
Comment (andreaskurth) [reply hw/ip/csrng/rtl/csrng_core.sv:1633]: Thanks for your feedback @vogelpi.

I have updated the description of the RTL fix to explain that this indeed fixes the code to implement a four-phase HS protocol.

I'd prefer to keep RTL and DV commit separate though, as one can then easily check that DV now detects this bug by reverting only the RTL commit. The two commits still mention each other in the description and are part of the same PR, so I think their relation should be clear.

[the remark this answers is not in the document]
Comment (vogelpi) [reply hw/ip/csrng/rtl/csrng_core.sv:1633]: Sounds good, thanks for changing this :+1:
